src/oData/jsonToOData.ts

```diff
@@ -215,18 +216,32 @@ export const convertToPineClientFilter = (
 	return handlePrimitiveFilter(parentKeys, filter);
 };
 
-export const orderbyBuilder = <T>(sortInfo: TableSortOptions<T> | null) => {
+export const orderbyBuilder = <T>(
+	sortInfo: TableSortOptions<T> | null,
+	customSort: AutoUIContext<T>['customSort'],
+) => {
 	if (!sortInfo) {
 		return null;
 	}
+
 	const { field, reverse, refScheme } = sortInfo;
 	if (!field) {
 		return null;
 	}
 	const direction = !reverse ? 'asc' : 'desc';
+	const customOrderByKey = customSort?.[field];
+	if (typeof customOrderByKey === 'string') {
+		return `${customOrderByKey} ${direction}`;
+	} else if (customOrderByKey != null && typeof customOrderByKey !== 'string') {
```

Comment:
I think this just needs to be v since we already checked for string:
```suggestion
	} else if (customOrderByKey != null) {
```